public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -136,15 +137,29 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
   }
 
   // Use this for tab completion features, wont publish response to other components
-  metadataRequest<T = any>(url: string) {
+  async metadataRequest<T = any>(url: string) {
     const data: any = {};
     for (const [key, value] of this.customQueryParameters) {
       if (data[key] == null) {
         data[key] = value;
       }
     }
 
-    return this._request<T>(url, data, { method: 'GET', hideFromInspector: true }).toPromise(); // toPromise until we change getTagValues, getTagKeys to Observable
+    // If URL includes endpoint that supports POST and GET method, try to use configured method. This might fail as POST is supported only in v2.10+.
+    if (GET_AND_POST_MEDATADATA_ENDPOINTS.some((endpoint) => url.includes(endpoint))) {
```

Comment:
how often do we do these? I think this can make some autocomplete features slower if you configure POST on older versions (which I assume is totally valid because you want POST for queries) and we have to do 2 requests for every metadata query. If we have to run this often maybe we still can cache the result so we then do only one correct request, would that make sense?

Reply:
This would be run in:
-  template variables (label_names() and label_values()) which are not cached, but they are either run just once, run when dashboard is initialised or when time picker changes. So not that often that it should slow down anything. 
- In autocomplete we are caching here labels and values, so shouldn't be problem as well.